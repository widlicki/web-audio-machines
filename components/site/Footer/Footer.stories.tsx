import { ComponentMeta, ComponentStory } from '@storybook/react'
import Footer from './Footer'

export default {
  title: 'components/site/Footer',
  component: Footer,
  parameters: {
    layout: 'fullscreen',
  },
} as ComponentMeta<typeof Footer>

const Template: ComponentStory<typeof Footer> = () => <Footer />

export const DefaultFooter = Template.bind({})
