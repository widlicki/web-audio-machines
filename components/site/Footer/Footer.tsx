import { Typography } from '@mui/material'
import Container from '@mui/material/Container'
import Grid from '@mui/material/Grid'
import theme from '../../../styles/theme'

const Footer = () => (
  <Container
    maxWidth={false}
    sx={{
      background: theme.background.color.light,
      borderTop: `4px solid ${theme.palette.secondary.main}`,
      color: 'common.white',
      height: '10vh',
    }}
  >
    <Grid container>
      <Grid
        item
        xs={12}
        sx={{
          textAlign: 'center',
          color: 'common.white',
          position: 'relative',
          marginTop: theme.spacing(3),
        }}
      >
        <Typography variant="body1">
          Â© web audio machines {new Date().getFullYear()}
        </Typography>
      </Grid>
    </Grid>
  </Container>
)

export default Footer
