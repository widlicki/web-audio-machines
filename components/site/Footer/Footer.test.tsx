import { render } from '@testing-library/react'
import Footer from './Footer'
// eslint-disable-next-line import/extensions
import { DefaultFooter } from './Footer.stories'

describe('Footer Stories', () => {
  new Map([['correct default footer', DefaultFooter]]).forEach((key) => {
    it(`renders ${key}`, () => {
      const component = <Footer />
      const { container } = render(component)
      expect(container).toMatchSnapshot()
    })
  })
})
