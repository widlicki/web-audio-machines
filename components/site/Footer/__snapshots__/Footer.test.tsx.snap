// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Stories renders function () { [native code] } 1`] = `
<div>
  <div
    class="MuiContainer-root css-1s55f6l-MuiContainer-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-12tubgu-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          Â© web audio machines 
          2022
        </p>
      </div>
    </div>
  </div>
</div>
`;
